src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -1092,3 +1118,43 @@ function getSessionId(): string {
 
     return sessionId;
 }
+
+async function createClientPipeTransport(
+    pipeName: string,
+    encoding: RAL.MessageBufferEncoding = 'utf-8'
+): Promise<PipeTransport> {
+    const connected = new Promise<[MessageReader, MessageWriter]>((resolve) => {
+        const server: net.Server = net.createServer((socket: net.Socket) => {
```

Comment:
What is the security on this pipe that's being created? Is it restricted to the user on the machine or is it more open?